### PR TITLE
kill g_powerManager global

### PR DESCRIPTION
**Pull request by lrusak**
This one was pretty simple. I've built and it runs.

~~One thing that is an issues is I wanted to init the CPowerManager in `initstagetwo` not `initstageone` however I get a segfault when SetDefaults() is called due to `CPowerManager` not being initialized yet~~
```
Thread 1 (Thread 0x7fe245a626c0 (LWP 6327)):
#0  0x0000000000ae53dc in CPowerManager::CanPowerdown() ()
#1  0x0000000000ae553b in CPowerManager::SetDefaults() ()
#2  0x0000000000cee688 in CSettings::InitializeDefaults() ()
#3  0x0000000000cedf76 in CSettings::InitializeDefinitions() ()
#4  0x0000000000cee279 in CSettings::Initialize() ()
#5  0x0000000000c441c5 in CApplication::Create(CAppParamParser const&) ()
#6  0x0000000000af60b7 in XBMC_Run ()
#7  0x000000000081781d in main ()
```
~~and the problem of having it in `initstageone` is that some of the logging is missed.~~

~~Is there a better solution? I'll try and look for one.~~

Solved it by moving the `SetDefaults()` call. Should it be in the `CPowerManager::Initialize()` instead?

`CPowerManager` is initilized a bit later now but I don't think there is any problems with that.

### CODE REVIEW

File: xbmc/platform/win32/WIN32Util.h [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "MediaSource.h"
 #include "guilib/Geometry.h"
 #include "powermanagement/PowerManager.h"
+#include "ServiceBroker.h"
```
Comment (garbear) [inline xbmc/platform/win32/WIN32Util.h]: This shouldn't be needed

Comment (lrusak) [reply xbmc/platform/win32/WIN32Util.h]: thanks, seems my script didn't work 100%

**Comment by Montellese** [comment]
You could try to do the `SetDefault()` call in the power manager during initialisation instead of doing in in `CSettings`.

**Comment by lrusak** [comment]
@Montellese yes I thought of that also :)

**Comment by Montellese** [comment]
Sorry just saw that you edited the PR description. I only read the initial e-mail :-/

**Comment by lrusak** [comment]
@Montellese no worries I appreciate any input! :smiley_cat:

**Comment by lrusak** [comment]
jenkins build this please

**Comment by notspiff** [comment]
please, do not expose a reference from manager, expose a pointer so broker can check if it is configured or not.

**Comment by lrusak** [comment]
@notspiff can you please elaborate? Don't all services do the same thing?

**Comment by notspiff** [comment]
Yes and it is just as horrid there, it is broken design. test env hits pmanager so i had to check if m_instance was non-null. That means that if u remove the global we need to be able to check if the powemanager itself has been instanced.

**Comment by lrusak** [comment]
@notspiff can we consider that outside the scope of this PR and clean that up with all the other services separately?

**Comment by akva2** [comment]
No because tests will break

**Comment by lrusak** [comment]
but jenkins is all green?

**Comment by notspiff** [comment]
after https://github.com/xbmc/xbmc/pull/13212 this is now fine for merge as far as i am concerned.

**Comment by lrusak** [comment]
thanks @notspiff !!! :smile_cat:

**Pull request merged**